src/hotspot/share/classfile/placeholders.cpp

```diff
@@ -132,9 +132,6 @@ void PlaceholderEntry::add_seen_thread(JavaThread* thread, PlaceholderTable::cla
   SeenThread* threadEntry = new SeenThread(thread);
   SeenThread* seen = actionToQueue(action);
 
-  assert(action != PlaceholderTable::LOAD_INSTANCE || !EnableWaitForParallelLoad || seen == nullptr,
-         "Only one LOAD_INSTANCE allowed at a time");
-
```

Comment:
Getting rid of the full assertion seems to go beyond obsoleting EnableWaitForParallelLoad.

Reply:
The assertion reduces to something somewhat useless.  Without EnableWaitForParallel load, you can have more then one thread with a LOAD_INSTANCE placeholder.  These threads will wait when the go call into ClassLoader.loadClass() now.
There is nothing to assert here anymore.

Reply:
Okay. I confess I don't fully grok this part though.

Reply:
The bootloader uses LOAD_INSTANCE to wait for other threads (JVM implementation of parallel capable class loading), but this function doesn't pass the class_loader so that would be the only useful modification to this assert.  But it seems too specific to actually catch any bugs.
Before this change with EnableWaitForParallelLoad, the non-parallel-capable class loader would wait on the LOAD_INSTANCE placeholder with double_lock_wait so there would never be more than one.  Now, both threads take out the placeholder, and are expected to wait in the ClassLoader.loadClass function.

Reply:
If -Xcomp stopped calling load_signature_classes, we could remove the LOAD_INSTANCE placeholder for non-parallel capable class loaders.  In the future.

Reply:
Okay. Thanks for the additional info.